src/main/java/com/codingame/core/AbstractPlayer.java

```diff
@@ -0,0 +1,63 @@
+package com.codingame.core;
+
+import java.util.List;
+
+public abstract class AbstractPlayer {
```

Comment:
Not sure to see why you provide an abstract class?

Reply:
We want to force the Game creator to extend it